MongoCk fails for old changelogs if mongodriver saved executionMillis as Integer instead of Long

## Description

When upgrading from Mongock 4.3.8 to version 5.0.35 on one of our stages we get the following error during Mongock run:

`io.mongock.api.exception.MongockException: java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.Long (java.lang.Integer and java.lang.Long are in module java.base of loader 'bootstrap') 
at io.mongock.runner.core.executor.MongockRunnerImpl.execute(MongockRunnerImpl.java:67) 
at io.mongock.runner.springboot.base.MongockInitializingBeanRunner.afterPropertiesSet(MongockInitializingBeanRunner.java:17) 
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863) 
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800) 
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620) 
....
at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88) Caused by: java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.Long (java.lang.Integer and java.lang.Long are in module java.base of loader 'bootstrap') 
at org.bson.Document.getLong(Document.java:266) 
at io.mongock.driver.mongodb.sync.v4.repository.MongoSync4ChangeEntryRepository.lambda$getEntriesLog$0(MongoSync4ChangeEntryRepository.java:117) 
at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195) 
at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655) 
at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) 
at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) 
at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913) 
at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) 
at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578) 
at io.mongock.driver.mongodb.sync.v4.repository.MongoSync4ChangeEntryRepository.getEntriesLog(MongoSync4ChangeEntryRepository.java:120) 
at io.mongock.driver.api.entry.ChangeEntryService.getEntriesMap(ChangeEntryService.java:67) 
at io.mongock.driver.api.entry.ChangeEntryService.getExecuted(ChangeEntryService.java:37) 
at io.mongock.runner.core.executor.operation.change.MigrationExecutorBase.loadExecutedChangeEntries(MigrationExecutorBase.java:424) 
at io.mongock.runner.core.executor.operation.change.MigrationExecutorBase.executeMigration(MigrationExecutorBase.java:92) 
at io.mongock.runner.core.executor.operation.change.MigrationExecutorBase.executeMigration(MigrationExecutorBase.java:47) 
at io.mongock.runner.core.executor.MongockRunnerImpl.execute(MongockRunnerImpl.java:53) ... 28 common frames omitted 
`

The reason is, that in some circumstances the mongo driver saves the executionMillis as Integer instead of Long.

## PRIORITY

NORMAL

## Version and environment

### Mongock
- Mongock version 5.0.35
- Spring Boot + Spring Data Mongodb
- builder approach

### Environment
- Spring Boot 2.6.9, Spring Data Mongodb 3.3.6
- Infrastructure: SO

## Steps to Reproduce

1. insert a change entry in the db where the executionMillis field is saved as Integer/INT32
2. run Mongock migration

## Behaviour

**Expected behavior:** The migration should run even if the mongo driver decided to save the executionMillis field as Integer/INT32

**Actual behavior:** MongoCk execution and Application run fails

**How often the bug happens:** Hard to say, in our case one of our test stages is affected. On the other stages executionMillis has been saved as Long/INT64.

## Additional context

I don't have any insights on when the mongo driver decides to save a value as Integer insteaf of Long, I only found this stackoverflow: https://stackoverflow.com/questions/47121602/mongodb-trying-to-read-long-from-json-causes-java-lang-integer-cannot-be-cast-t

--- a/drivers/mongodb/mongodb-driver-test-template/src/main/java/io/mongock/driver/mongodb/test/template/MongoChangeEntryRepositoryITestBase.java
+++ b/drivers/mongodb/mongodb-driver-test-template/src/main/java/io/mongock/driver/mongodb/test/template/MongoChangeEntryRepositoryITestBase.java
@@ -2,6 +2,7 @@ package io.mongock.driver.mongodb.test.template;
 
 import com.mongodb.client.model.IndexOptions;
 import io.mongock.api.exception.MongockException;
+import io.mongock.driver.api.entry.ChangeEntry;
 import io.mongock.driver.api.entry.ChangeEntryService;
 import io.mongock.driver.mongodb.test.template.util.IntegrationTestBase;
 import org.bson.Document;
@@ -9,6 +10,7 @@ import org.junit.jupiter.api.Test;
 
 import java.time.Instant;
 import java.util.Date;
+import java.util.List;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
@@ -33,15 +35,23 @@ public abstract class MongoChangeEntryRepositoryITestBase extends IntegrationTes
     initializeRepository(false);
   }
 
-
-
-  private void createAndInsertChangeEntry(boolean withState, String state, String changeId, String author, String executionId) {
+  @Test
+  public void getEntriesLog_WorksRegardlessOfExecutionMillisMissingIntegerOrLong() throws MongockException {
     initializeRepository(true);
+    createAndInsertChangeEntry(false, null, "changeId1", "author", "executionId1", Long.MAX_VALUE);
+    createAndInsertChangeEntry(false, null, "changeId2", "author", "executionId2", Integer.MAX_VALUE);
+    createAndInsertChangeEntry(false, null, "changeId3", "author", "executionId3", null);
+    List<ChangeEntry> result = repository.getEntriesLog();
+    assertEquals(3, result.size());
+  }
+
+  private void createAndInsertChangeEntry(boolean withState, String state, String changeId, String author, String executionId, Number executionMillis) {
     Document existingEntry = new Document()
         .append("executionId", executionId)
         .append("changeId", changeId)
         .append("author", author)
         .append("timestamp", Date.from(Instant.now()))
+        .append("executionMillis", executionMillis)
         .append("changeLogClass", "anyClass")
         .append("changeSetMethod", "anyMethod")
         .append("metadata", null);

--- a/drivers/mongodb/mongodb-reactive-driver/src/main/java/io/mongock/driver/mongodb/reactive/repository/MongoReactiveChangeEntryRepository.java
+++ b/drivers/mongodb/mongodb-reactive-driver/src/main/java/io/mongock/driver/mongodb/reactive/repository/MongoReactiveChangeEntryRepository.java
@@ -111,7 +111,7 @@ public class MongoReactiveChangeEntryRepository extends MongoReactiveRepositoryB
             entry.containsKey(KEY_TYPE) ? ChangeType.valueOf(entry.getString(KEY_TYPE)) : null,
             entry.getString(KEY_CHANGELOG_CLASS),
             entry.getString(KEY_CHANGESET_METHOD),
-            entry.containsKey(KEY_EXECUTION_MILLIS) ? entry.getLong(KEY_EXECUTION_MILLIS) : -1L,
+            entry.containsKey(KEY_EXECUTION_MILLIS) ? ((Number) entry.get(KEY_EXECUTION_MILLIS)).longValue() : -1L,
             entry.getString(KEY_EXECUTION_HOSTNAME),
             entry.get(KEY_METADATA)))
         .collect(Collectors.toList());

--- a/drivers/mongodb/mongodb-sync-v4-driver/src/main/java/io/mongock/driver/mongodb/sync/v4/repository/MongoSync4ChangeEntryRepository.java
+++ b/drivers/mongodb/mongodb-sync-v4-driver/src/main/java/io/mongock/driver/mongodb/sync/v4/repository/MongoSync4ChangeEntryRepository.java
@@ -114,7 +114,7 @@ public class MongoSync4ChangeEntryRepository extends MongoSync4RepositoryBase<Ch
             entry.containsKey(KEY_TYPE) ? ChangeType.valueOf(entry.getString(KEY_TYPE)) : null,
             entry.getString(KEY_CHANGELOG_CLASS),
             entry.getString(KEY_CHANGESET_METHOD),
-            entry.containsKey(KEY_EXECUTION_MILLIS) ? entry.getLong(KEY_EXECUTION_MILLIS) : -1L,
+            entry.containsKey(KEY_EXECUTION_MILLIS) ? ((Number) entry.get(KEY_EXECUTION_MILLIS)).longValue() : -1L,
             entry.getString(KEY_EXECUTION_HOSTNAME),
             entry.get(KEY_METADATA)))
         .collect(Collectors.toList());

--- a/drivers/mongodb/mongodb-v3-driver/src/main/java/io/mongock/driver/mongodb/v3/repository/Mongo3ChangeEntryRepository.java
+++ b/drivers/mongodb/mongodb-v3-driver/src/main/java/io/mongock/driver/mongodb/v3/repository/Mongo3ChangeEntryRepository.java
@@ -111,7 +111,7 @@ public class Mongo3ChangeEntryRepository extends Mongo3RepositoryBase<ChangeEntr
             entry.containsKey(KEY_TYPE) ? ChangeType.valueOf(entry.getString(KEY_TYPE)) : null,
             entry.getString(KEY_CHANGELOG_CLASS),
             entry.getString(KEY_CHANGESET_METHOD),
-            entry.containsKey(KEY_EXECUTION_MILLIS) ? entry.getLong(KEY_EXECUTION_MILLIS) : -1L,
+            entry.containsKey(KEY_EXECUTION_MILLIS) ? ((Number) entry.get(KEY_EXECUTION_MILLIS)).longValue() : -1L,
             entry.getString(KEY_EXECUTION_HOSTNAME),
             entry.get(KEY_METADATA)))
         .collect(Collectors.toList());

